src/libs/actions/Report.js

```diff
@@ -1225,6 +1157,107 @@ Onyx.connect({
     callback: handleReportChanged,
 });
 
+/**
+ * Deletes a comment from the report, basically sets it as empty string
+ *
+ * @param {Number} reportID
+ * @param {Object} reportAction
+ */
+function deleteReportComment(reportID, reportAction) {
+    const sequenceNumber = reportAction.sequenceNumber;
+
+    const optimisticReportActions = {
+        [sequenceNumber]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+        },
+    };
+    const optimisticReport = {
+        lastMessageText: ReportActions.getLastVisibleMessageText(reportID, {
+            [sequenceNumber]: {
+                ...reportAction,
+            },
+        }),
+    };
+
+    // If the deleted comment is more recent than our last read comment, update the unread count
+    if (sequenceNumber > getLastReadSequenceNumber(reportID)) {
+        optimisticReport.unreadActionCount = Math.max(getUnreadActionCount(reportID) - 1, 0);
+    }
+
+    // List of optmistic data to change
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: optimisticReportActions,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: optimisticReport,
+        },
+    ];
+
+    // On Success remove the message content and pendingAction state
+    const successData = {
+        [sequenceNumber]: {
+            pendingAction: null,
+            message: [{
+                html: '',
+                text: '',
+            }],
+        },
+    };
+
+    // On Error clear the pendingAction state and revert the message
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: {
+                [sequenceNumber]: {
+                    message: reportAction.message,
+                    pendingAction: null,
+                },
+            },
+        },
+
+        // {
+        //     onyxMethod: CONST.ONYX.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+        //     value: {
+        //         lastMessageText:
+        //         unreadActionCount:
+        //     }
+        // }
```

Comment:
Ok so if the request fails we would technically want to put back the last visible message text from the current set of actions like this plus the original action we will revert to: 

```js
ReportActions.getLastVisibleMessageText(reportID, {
    [sequenceNumber]: reportAction,
});
```

and same as before but reverse if `sequenceNumber > getLastReadSequenceNumber(reportID)` + 1 on the `unreadActionCount`. We are getting rid of `unreadActionCount` very soon so I am not concerned with this approach.

Reply:
As we discussed, we'll leave the server responsible for updating the lastMessageText in case of any failures in the deletion and update!